File: coding-test-1/practice-problems/problems/02-longest-word.rb
# Write a method that takes in a string. Return the longest word in
# the string. You may assume that the string contains only letters and
# spaces. If there are multiple longest words, return the first one
# that appears in the sentence.
#
# You may use the String `split` method to aid you in your quest.
#
# Difficulty: easy.

def longest_word(sentence)
end

# These are tests to check that your code is working. After writing
# your solution, they should all print true.

puts(
  'longest_word("short longest") == "longest": ' +
  (longest_word('short longest') == 'longest').to_s
)
puts(
  'longest_word("one") == "one": ' +
  (longest_word('one') == 'one').to_s
)
puts(
  'longest_word("abc def abcde") == "abcde": ' +
  (longest_word('abc def abcde') == 'abcde').to_s
)
puts(
'longest_word("aa one two bb ten") == "one": ' +
(longest_word('aa one two bb ten') == 'one').to_s
)

File: coding-test-1/practice-problems/solutions-python/02-longest-word.py
# Write a method that takes in a string. Return the longest word in
# the string. You may assume that the string contains only letters and
# spaces.
#
# You may use the String `split` method to aid you in your quest.
#
# Difficulty: easy.
import unittest


def longest_word(sentence):
    words = sentence.split()
    current_longest = None

    for word in words:
        if current_longest is None:
            current_longest = word
        elif len(word) > len(current_longest):
            current_longest = word

    return current_longest

# These are tests to check that your code is working. After writing
# your solution, they should all run.


class LongestWord(unittest.TestCase):

    def test_two_words(self):
        self.assertEqual(longest_word("short longest"), "longest")

    def test_one_word(self):
        self.assertEqual(longest_word("one"), "one")

    def test_three_words(self):
        self.assertEqual(longest_word("short longest"), "longest")

if __name__ == '__main__':
    unittest.main()